WooCommerce/Classes/Notifications/PushNotificationBackgroundSynchronizer.swift

```diff
@@ -27,22 +27,27 @@ struct PushNotificationBackgroundSynchronizer {
     ///
     @MainActor
     func sync() async -> UIBackgroundFetchResult {
-
         guard let pushNotification = PushNotification.from(userInfo: userInfo) else {
             return .noData
         }
 
         do {
-
             let startTime = Date.now
-
-            // I'm not sure why we need to sync all notifications instead of only the current one.
-            // This is legacy code copied from PushNotificationsManager
-            try await synchronizeNotifications()
-
-            // Find the orderID from the previously synced notification.
-            guard let orderID = getOrderID(noteID: pushNotification.noteID) else {
-                return .newData
+            let orderID: Int64
+
+            if let note = pushNotification.note,
+               let noteOrderID = note.meta.identifier(forKey: .order) {
+                orderID = Int64(noteOrderID)
+            } else {
```

Comment:
This the only change, checking if we already have the orderID in the notification `note` property.